core-kotlin-modules/core-kotlin-concurrency-3/src/main/java/com/baeldung/errorhandling/ExceptionHandlingMethods.kt

```diff
@@ -0,0 +1,60 @@
+package com.baeldung.errorhandling
+
+import kotlinx.coroutines.*
+
+fun main() = runBlocking {
+    launch {
+        try {
+            val result = 10 / 0
+            println("Result: $result")
+        } catch (e: ArithmeticException) {
+            println("Caught an ArithmeticException: $e")
+        }
+    }
+    delay(1000)
+}
+
+
+fun main1() = runBlocking {
+    try {
+        coroutineScope {
+            launch {
+                delay(200)
+                throw CustomException("An exception occurred!")
+            }
+            launch {
+                delay(100)
+                println("This coroutine completes successfully.")
+            }
+        }
+    } catch (e: CustomException) {
+        println("Caught exception: ${e.message}")
+    }
+}
+
+class CustomException(message: String) : Exception(message)
+
+fun main2() = runBlocking {
+    val exceptionHandler = CoroutineExceptionHandler { _, exception ->
+        println("Caught an exception: ${exception.message}")
+    }
+    supervisorScope {
+        val job = launch(exceptionHandler) {
+            launch { delay(200); throw CustomException("An exception occurred!") }
+            launch { delay(100); println("This coroutine completes successfully.") }
+        }
```

Comment:
Apologies if I missed this before, but let's make these multi-lines.

```suggestion
        val job = launch(exceptionHandler) {
            launch {
                delay(200)
                throw CustomException("An exception occurred!")
            }
            launch {
                delay(100)
                println("This coroutine completes successfully.")
            }
        }
```